resolwe_bio/processes/differential_expression/dexseq.yml

```diff
@@ -0,0 +1,134 @@
+# ================================
+# Differential Expression - DEXSeq
+# ================================
+---
+
+- slug: differentialexpression-dexseq
+  name: DEXSeq
+  requirements:
+    expression-engine: jinja
+    executor:
+      docker:
+        image: resolwebio/rnaseq:2.1.1
+  data_name: 'Differential expression (case vs. control)'
+  version: 0.0.1
+  type: data:differentialexpression:dexseq
+  category: analyses
+  persistence: CACHED
+  description: >
+    Find differentially expressed exons.
+  input:
+    - name: case
+      label: Case
+      type: list:data:expression
+      description: >
+        Case samples (replicates).
+    - name: control
+      label: Control
+      type: list:data:expression
+      description: >
+        Control samples (replicates).
+    - name: annotation
+      label: Annotation GFF
+      type: data:annotation
+      description: >
+        DEXSeq readable annotation file.
+  output:
+    - name: raw 
+      label: Differential expression
+      type: basic:file
+    - name: dxr
+      label: DEXSeq object
+      type: basic:file
+    - name: source
+      label: Gene ID source
+      type: basic:string
+    - name: species
+      label: Species
+      type: basic:string
+    - name: build
+      label: Build
+      type: basic:string
+    - name: feature_type
+      label: Feature type
+      type: basic:string
+  run:
+    runtime: polyglot
+    language: bash
+    program: |
+      {% set expressions = control + case %}
+
+      {% for e in expression %}
+
+        {% if e.source != (expressions|first).source %}
+          re-warning "All expression data must be annotated by the same genome database."
+          re-error "Expression {{ expressions|first|name }} has {{ (expressions|first).source }} gene IDs, while {{ e|name }} has {{ e.source }} gene IDs."
+        {% endif %}
+
+        {% if e.species != (expressions|first).species %}
+          re-warning "All expressions must be of the same species."
+          re-error "Expression {{ expressions|first|name }} is {{ (expressions|first).species }}, while {{ e|name }} is {{ e.species }}."
```

Comment:
Why are we issuing a warning and then throwing an error?

Reply:
@romunov  this was a part of the error messages rehaul that @JenkoB made so she is probably best suited to answer this question.

@mzganec I agree this is not the best way. However this does not return sample names but data object names. But yes, these are often the same. I'll look into it and come back with something better. We should have a discussion on rehauling error messages again and come up with a good practice that we can then use everywhere.

Reply:
An easy semantic fix:
re-warning: "All **samples** must be the same species."
re-errorr: "The samples `Sample1_name` is from `organism1` while `Sample2_name` is from `organism2`.

Reply:
I agree with Rafes suggestion, and it would be good to correct it in other processes. The decision of giving Warning and Error is because it displayed better having it that way than error in more lines... I am not committed to this decisions, so I support all improvements. :)